addons/website/static/src/snippets/s_disclaimer/options.js

```diff
@@ -0,0 +1,40 @@
+import options from "@web_editor/js/editor/snippets.options";
+
+options.registry.Disclaimer = options.Class.extend({
+    displayOverlayOptions: true,
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Moves the snippet in #o_snippet_above_header to be common to all pages or inside
+     * the first editable oe_structure in the main to be on current page only.
+     *
+     */
+    moveBlock(previewMode, widgetValue, params) {
+        const snippetEl = this.$target[0].closest(".s_disclaimer");
+        let whereEl;
+        if (widgetValue === "allPages") {
+            whereEl = this.ownerDocument.querySelector("#o_snippet_above_header");
+        } else {
+            whereEl = this.ownerDocument.querySelector("#wrap");
+        }
```

Comment:
Hello @msh-odoo both ways are okay for me, but I prefer using an if-else ladder here for future scope. If any condition-specific code needs to be added later, it will avoid extra differences.